frontend/src/pages/index.tsx

```diff
@@ -1,8 +1,68 @@
 /* eslint-disable canonical/filename-match-exported */
 import { type NextPage } from 'next';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+
+// components
+import Modal from '@/components/modal';
+
+// styles
+const Layout = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  height: 120vh; // to test scroll
+`;
+
+const ModalContent = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const Button = styled.button`
+  background: #9f7aea;
+  border-radius: 8px;
+  border: transparent;
+  cursor: pointer;
+  color: white;
+  padding: 8px 16px;
+  transition: all 0.1s ease-in;
+  width: fit-content;
+
+  &:hover {
+    background: #44337a;
+  }
+`;
 
 const Index: NextPage = () => {
-  return <h1>Welcome to Contra!</h1>;
+  const [open, setOpen] = useState(false);
+
+  // may be overkill, but reducing re-renders as we have a couple useEffects in the modal
+  const openHandler = useCallback(() => {
+    setOpen(true);
+  }, []);
+
+  const closeHandler = useCallback(() => {
+    setOpen(false);
+  }, []);
```

Comment:
Want to preface that preemptive performance changes are an anti-pattern, and these changes usually should be made after doing an internal investigation/audit (via 3rd party software like lighthouse).